src/health_checker.cc

```diff
@@ -35,6 +36,9 @@ void HealthChecker::SetUnhealthy(const std::string& component) {
   unhealthy_components_.insert(component);
   std::ofstream health_file(config_.HealthCheckFile());
   health_file << std::endl;
+  if (config_.KillAgentOnFailure()) {
+    std::exit(EXIT_FAILURE);
```

Comment:
Should we change `health_checker_unittest` to also test this behavior? See [`EXPECT_EXIT`](https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#death-tests).

Reply:
Added a test, thanks for the link